File: test/test_files/java/com/example/core/DataProcessor.java
package com.example.core;

import com.example.utils.Logger;
import com.example.utils.Helper;
import java.util.HashMap;
import java.util.Map;

/**
 * Concrete processor for data handling.
 */
public class DataProcessor extends BaseProcessor {
    private int processedCount;
    
    public DataProcessor(String name) {
        super(name);
        this.processedCount = 0;
    }
    
    @Override
    public String process(Object data) {
        logger.log("Processing data: " + data);
        
        // Control statement: if/else for validation
        if (data == null || data.toString().isEmpty()) {
            throw new IllegalArgumentException("Data cannot be empty");
        }
        
        // Control statement: try/catch for validation
        try {
            Helper.validateInput(data);
        } catch (Exception e) {
            logger.log("Validation failed: " + e.getMessage());
            throw e;
        }
        
        String result = Helper.formatOutput(data);
        processedCount++;
        return result;
    }
    
    public Map<String, Object> getStats() {
        Map<String, Object> stats = new HashMap<>();
        stats.put("name", name);
        stats.put("processed", processedCount);
        
        // Control statement: if/elif/else for status
        if (processedCount == 0) {
            stats.put("status", "idle");
        } else if (processedCount < 10) {
            stats.put("status", "active");
        } else {
            stats.put("status", "busy");
        }
        
        return stats;
    }
}

File: test/test_files/cpp/src/core/data_processor.cpp
#include "../../include/core/data_processor.h"
#include "../../include/utils/helper.h"

namespace core {

DataProcessor::DataProcessor(const std::string& name) 
    : BaseProcessor(name), processedCount(0) {}

std::string DataProcessor::process(const std::string& data) {
    logger->log("Processing data: " + data);
    
    // Control statement: if/else for validation
    if (data.empty()) {
        throw std::invalid_argument("Data cannot be empty");
    }
    
    // Control statement: try/catch for validation
    try {
        utils::Helper::validateInput(&data);
    } catch (const std::exception& e) {
        logger->log(std::string("Validation failed: ") + e.what());
        throw;
    }
    
    std::string result = utils::Helper::formatOutput(data);
    processedCount++;
    return result;
}

std::map<std::string, int> DataProcessor::getStats() {
    std::map<std::string, int> stats;
    stats["processed"] = processedCount;
    
    // Control statement: if/else if/else for status code
    int statusCode;
    if (processedCount == 0) {
        statusCode = 0;  // idle
    } else if (processedCount < 10) {
        statusCode = 1;  // active
    } else {
        statusCode = 2;  // busy
    }
    stats["status_code"] = statusCode;
    
    return stats;
}

} // namespace core
